react/features/token-data/middleware.js

```diff
@@ -0,0 +1,69 @@
+import { SET_ROOM_URL } from '../app';
+import { getConfigParamsFromUrl } from '../base/config';
+import { MiddlewareRegistry } from '../base/redux';
+
+import { setTokenData } from './actions';
+import { TokenData } from './TokenData';
+
+/**
+ * Middleware that parsing token data after setting new room URL.
+ *
+ * @param {Store} store - The Redux store.
+ * @returns {Function}
+ * @private
+ */
+MiddlewareRegistry.register(store => next => action => {
+    switch (action.type) {
+    case SET_ROOM_URL:
+        return _parseTokenData(store, next, action);
+    }
+
+    return next(action);
+});
+
+/**
+ * Parses token data from room URL and if it's found stores into Redux
+ * store.
+ *
+ * @param {Store} store - The Redux store in which the specified action is being
+ * dispatched.
+ * @param {Dispatch} next - The Redux dispatch function to dispatch the
+ * specified action to the specified store.
+ * @param {Action} action - The Redux action SET_ROOM_URL which is being
+ * dispatched in the specified store.
+ * @returns {Object} The new state that is the result of the reduction of the
+ * specified action.
+ * @private
+ */
+function _parseTokenData(store, next, action) {
+    const nextState = next(action);
```

Comment:
can't you get the config from the current state and the room URL from the action? I don't like depending on the next state very much here, looks fragile.